crates/pallet-domains/src/block_tree.rs

```diff
@@ -253,22 +253,37 @@ pub(crate) fn process_execution_receipt<T: Config>(
                 domain_block_number.checked_sub(&T::BlockTreePruningDepth::get())
             {
                 let receipts_at_number = BlockTree::<T>::take(domain_id, to_prune);
+
+                // The receipt at `to_prune` may already been pruned if there is fraud proof being
+                // processed and the `HeadReceiptNumber` is reverted.
+                if receipts_at_number.is_empty() {
+                    return Ok(None);
+                }
+
+                // FIXME: the following check is wrong because the ER at `to_prune` may not necessarily go through
```

Comment:
If the to_prune ER is submitted in this block which at this point it finlized, we should not accept that bundle. Am I missing something here ?

Reply:
If an ER is already finalized/pruned then subsequent submission of the same ER will be rejected, but the point here is 
the block tree already contains ER of `#1..#10` and another (fraudulent) ER of `#1` is submitted, then both the valid and fraudulent ERs of `#1` are finalized by the submission of ER `#11`, thus the fraudulent ER only open for challenging in a small period of time.

Reply:
I understand that. What I'm suggesting is that #1 should not be accepted since it will be finalized in this block since, like you said, there wont be enough time for operators to submit a fraud proof

Reply:
What you saying is any ER between `#1..#10` should not be accepted and that is what #1731 suggested. But under the current implementation, as the comment points out, the ER that should be considered as finalized is the one that has `BlockTreePruningDepth` number of ERs built on top of it instead of any ER that has the same `domain_block_number`.

Reply:
> What you saying is any ER between #1..#10 should not be accepted and that is w

Nope, I was suggesting only #1 should not be accepted in the block in which it will be finalized. This will be moot when #1731 so I guess this is fine for now

Reply:
It doesn't make much difference if the attacker submits an invalid ER of `#2` then this ER is only open for challenging in 2 blocks which is still much smaller than the expected challenge period.